internal/factsengine/gatherers/hostsfile_test.go

```diff
@@ -0,0 +1,114 @@
+package gatherers
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type HostsFileTestSuite struct {
+	suite.Suite
+	fileSystem afero.Fs
+}
+
+func TestHostsFileTestSuite(t *testing.T) {
+	suite.Run(t, new(HostsFileTestSuite))
+}
+
+func (suite *HostsFileTestSuite) SetupTest() {
+	suite.fileSystem = afero.NewMemMapFs()
+
+	err := suite.fileSystem.MkdirAll("/etc", 0644)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (suite *HostsFileTestSuite) TestHostsFileBasic() {
+	testFile, _ := os.Open("../../../test/fixtures/gatherers/hosts.basic")
+	confFile, _ := ioutil.ReadAll(testFile)
+	err := afero.WriteFile(suite.fileSystem, "/etc/hosts", confFile, 0644)
+	assert.NoError(suite.T(), err)
+	c := NewHostsFileGatherer(suite.fileSystem)
+
+	factRequests := []FactRequest{
+		{
+			Name:     "hosts_localhost",
+			Gatherer: "hosts",
+			Argument: "localhost",
```

Comment:
I would put the `CheckID` in the `FactRequest` entry, so we test that the same `CheckID` is present in the result